### ISSUE TITLE
CDAT Migration Phase 2: Refactor `enso_diags` set

### ISSUE THREAD

**Issue by tomvothecoder** [E3SM-Project/e3sm_diags#663]
Just noting that @forsyth2 previously worked on this set.

**Comment by forsyth2** [comment]
Initial analysis of refactoring needs:

# Direct dependency analysis
Per the direct dependency analysis at #803:
```
enso_diags

git grep -n cdms2 e3sm_diags/driver/enso_diags_driver.py
e3sm_diags/driver/enso_diags_driver.py:8:import cdms2
e3sm_diags/driver/enso_diags_driver.py:238:            with cdms2.open(mask_path) as f:

git grep -n cdutil e3sm_diags/driver/enso_diags_driver.py
e3sm_diags/driver/enso_diags_driver.py:9:import cdutil
e3sm_diags/driver/enso_diags_driver.py:96:        sst_avg = cdutil.averager(sst_nino, axis="xy")
e3sm_diags/driver/enso_diags_driver.py:98:        sst_avg_anomaly = cdutil.ANNUALCYCLE.departures(sst_avg)
e3sm_diags/driver/enso_diags_driver.py:127:    cdutil.setTimeBoundsMonthly(domain)
e3sm_diags/driver/enso_diags_driver.py:131:    anomaly = cdutil.ANNUALCYCLE.departures(domain)
e3sm_diags/driver/enso_diags_driver.py:135:    confidence_levels = cdutil.ANNUALCYCLE.departures(domain)[0, :, :](squeeze=1)
e3sm_diags/driver/enso_diags_driver.py:424:            test_avg = cdutil.averager(test_data_ts_regional, axis="xy")
e3sm_diags/driver/enso_diags_driver.py:425:            ref_avg = cdutil.averager(ref_data_ts_regional, axis="xy")
e3sm_diags/driver/enso_diags_driver.py:427:            y["test"] = cdutil.ANNUALCYCLE.departures(test_avg)
e3sm_diags/driver/enso_diags_driver.py:428:            y["ref"] = cdutil.ANNUALCYCLE.departures(ref_avg)

git grep -n cdutil e3sm_diags/plot/cartopy/enso_diags_plot.py
e3sm_diags/plot/cartopy/enso_diags_plot.py:6:import cdutil
e3sm_diags/plot/cartopy/enso_diags_plot.py:87:        domain = cdutil.region.domain(latitude=(-90.0, 90, "ccb"))

git grep -n genutil e3sm_diags/driver/enso_diags_driver.py
e3sm_diags/driver/enso_diags_driver.py:146:            # Uncomment the following line to use CDAT/genutil instead
e3sm_diags/driver/enso_diags_driver.py:148:            # slope, intercept = genutil.statistics.linearregression(dependent_var, x=independent_var)
```

# Driver

https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/driver/enso_diags_driver.py#L96-L98
=> the 2 `cdutil` references need to be changed out.

https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/driver/enso_diags_driver.py#L127-L135
=> the 3 `cdutil` references need to be changed out.

https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/driver/enso_diags_driver.py#L146-L149
=> just remove this comment block

https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/driver/enso_diags_driver.py#L169-L176
=> these functions are defined in https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/metrics/__init__.py. `max_cdms`, `min_cdms` don't appear to actually call anything in CDAT. But `mean` uses `cdutil` and `std` uses `genutil`.

https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/driver/enso_diags_driver.py#L188-L189
=> these functions are defined in https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/metrics/__init__.py. `rmse` and `corr`use `genutil`.

https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/driver/enso_diags_driver.py#L202-L203
=> https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/driver/utils/dataset.py has a number of CDAT dependencies.

https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/driver/enso_diags_driver.py#L230-L240
=> `ds_land_sea_mask: xr.Dataset` = test_ds._get_land_sea_mask(season) (as in https://github.com/E3SM-Project/e3sm_diags/pull/749/files#r1467834914). This is important because `get_climo_variable` calls `_get_climo_var` which uses `cdms2`.

https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/driver/enso_diags_driver.py#L424-L428
=> the 4 `cdutil` references need to be changed out.

# Plot
I'll do an initial refactoring analysis for plotting in a separate comment.

**Comment by forsyth2** [comment]
https://github.com/E3SM-Project/e3sm_diags/discussions/803#discussioncomment-9064323 may prove useful in finding CDAT dependencies that come in via function calls.

**Comment by forsyth2** [comment]
# Plot

https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/plot/cartopy/enso_diags_plot.py#L28-L32
=> Rename to `PANEL_CFG`

https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/plot/cartopy/enso_diags_plot.py#L36
=> Rename to `BORDER_PADDING`

The two changes above will be passed into:
https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/plot/utils.py#L59

https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/plot/cartopy/enso_diags_plot.py#L67-L69
=> I'm not entirely certain based on visual inspection. But it looks like any method of `var` is going to ultimately be from CDAT. `var`. E.g., `var` is `test` in the first call to `plot_panel_map` inside `plot_map`. `test` is `test_reg_coe` in the call to `plot_map` (in the driver). `test_reg_coe` is defined by `perform_regression`. In `perform_regression`, it is `reg_coe` which is defined as `anomaly[0, :, :](squeeze=1)`, and `anomaly` is defined as `anomaly = cdutil.ANNUALCYCLE.departures(domain)`.

https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/plot/cartopy/enso_diags_plot.py#L314-L341
=> Replace with a call to https://github.com/E3SM-Project/e3sm_diags/blob/f6c4fdfeabb6765c330b93bee8d18063efb1f9a6/e3sm_diags/plot/utils.py#L59

**Comment by tomvothecoder** [comment]
Closed by #832
